odoo/fields.py

```diff
@@ -3083,7 +3098,7 @@ class Many2one(_Relational):
     auto_join = False                   # whether joins are generated upon search
     delegate = False                    # whether self implements delegation
 
-    def __init__(self, comodel_name=Default, string=Default, **kwargs):
+    def __init__(self, comodel_name: _type[_T] = Default, string: str = Default, **kwargs):
```

Comment:
You need the type of the default value too. It's a little bit hard to do in python, I ended up with something like:

```python
class DefaultType(Enum): DEFAULT = "_"

Default = DefaultType.DEFAULT
```